examples/passport-login/src/authentication-strategies/oauth2.ts

```diff
@@ -0,0 +1,123 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: example-passport-oauth2-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  AuthenticationStrategy,
+  UserIdentityService,
+  asAuthStrategy,
+  AuthenticationBindings,
+} from '@loopback/authentication';
+import {StrategyAdapter} from '@loopback/authentication-passport';
+import {Profile} from 'passport';
+import {Strategy, StrategyOptions} from 'passport-oauth2';
+import {Request, RedirectRoute} from '@loopback/rest';
+import {UserProfile, securityId} from '@loopback/security';
+import {User} from '../models';
+import {UserServiceBindings} from '../services';
+import {inject, bind, extensions, Getter} from '@loopback/core';
+
+export namespace PassportAuthenticationBindings {
+  export const OAUTH2_STRATEGY = 'passport.authentication.oauth2.strategy';
+}
+
+@bind(asAuthStrategy)
+export class Oauth2AuthStrategy implements AuthenticationStrategy {
```

Comment:
I think this class needs to declare the extension point

@extensionPoint(PassportAuthenticationBindings.OAUTH2_STRATEGY)  . see my comment above.

Reply:
@emonddr @extensionPoint is just an addition to @extensions() decorator. It is optional. If you test the given application , you will see that the app works fine without it.

Reply:
https://loopback.io/doc/en/lb4/core-tutorial-part5.html#access-extensions-for-a-given-extension-point   uses @extensionPoint on the class that searches for @extensions. 

I don't see it mention that @extensionPoint is optional. But I do see this sentence:
```
By default, the name of the extension point is the same as the class name. To customize metadata such as name for the extension point, we can use @extensionPoint to decorate the class
```
So does this mean that any class bound to the context (class doesn't use @extensionPoint) is considered an extension point. So if I have N different classes that I am binding to a context (that have nothing to do with the extensionPoint/extension pattern, they are automatically considered an extensionPoint? I thought that having the @extensionPoint decorator whas to give a particular class the unique status of being an actual extension point to which someone can register extensions. @raymond, is @extensionPoint really optional?

Reply:
@emonddr I dont have much knowledge on how the metadata is organized differently, by using `@extensionpoint` and the docs dont seem to clarify that as well. @raymondfeng should be able to answer that. But the code works for this example.

Reply:
@emonddr The auth provider should be decorated with `@extensionPoint`, see example in https://loopback.io/doc/en/lb4/Loopback-component-authentication.html#registering-a-custom-authentication-strategy

The class here is just an auth strategy.

Reply:
@emonddr @deepakrkris See the comment in https://github.com/strongloop/loopback-next/blob/master/packages/core/src/extension-point.ts#L67-L71

The extension name can be provided directly by the 1st param in `@extensions()`, `@extensionPoint()` is another way to provide the name.
That's why Deepak's code works.